test/e2e/specs/fse/fse__temp-smoke-test.ts

```diff
@@ -23,14 +23,27 @@ declare const browser: Browser;
  *
  * The goal here is to catch major breaks with the integration --- i.e. Calypso navigation no long working,
  * or getting a WSOD when trying to load the editor.
+ *
+ *
+ * Keywords: FSE, Full Site Editor, Gutenberg
  */
-
 describe( DataHelper.createSuiteTitle( 'Site Editor Smoke Test' ), function () {
 	let page: Page;
+	let testAccount: TestAccount;
 	let fullSiteEditorPage: FullSiteEditorPage;
 
-	const features = envToFeatureKey( envVariables );
-	const accountName = getTestAccountByFeature( { ...features, variant: 'siteEditor' }, [
+	// Jetpack deploy (`atomicVariation`) and "regular" Calypso E2E have to use different sets of
+	// feature keys that are not mutually compatible.
+	// Regular and CoBlocks runs must override the feature keys to add the `variant: siteEditor` clause.
+	// Meanwhile, for Jetpack deploys the extra `variant` causes `getTestAccountByFeature` to report
+	// no valid match because none of the Jetpack deploy test users in the default criteria have
+	// `variant` keys.
+	let features = envToFeatureKey( envVariables );
+	if ( envVariables.JETPACK_TARGET !== 'wpcom-deployment' ) {
+		features = { ...features, variant: 'siteEditor' };
```

Comment:
I'm totally open to a better, more intuitive way to override the feature keys here! 

I spent some time trying to understand how `envToFeatureKey` and `getTestAccountByFeature` interact with the default criteria and it only kept getting more confusing.

Through experimenting I found the `variant: 'siteEditor'` addition to the feature key was interfering with the test account matching in Jetpack E2E scenario. I decided to key off the one environment variable that changes between "normal" Calypso runs and Jetpack runs.

@dpasque - I'm definitely open to better ideas here.

Reply:
Gotcha. It did occur to me that I should upgrade the default matrix instead, and not sure why I didn't do that. Probably because I tried to stay away from the whole `getTestAccountByFeature` side of things.